internal/component/azure/servicebus/subscription.go

```diff
@@ -197,6 +202,15 @@ func (s *Subscription) handleAsync(ctx context.Context, msg *azservicebus.Receiv
 	go func() {
 		var consumeToken bool
 		var err error
+		var messageKey int64
+
+		// We check for the existence of the sequence number again just in case
+		// it has been unset by some other goroutine and we cannot process the message.
+		if msg.SequenceNumber == nil {
+			s.logger.Errorf("Message has no sequence number: %s", msg.MessageID)
+			s.AbandonMessage(ctx, msg)
+		}
+		messageKey = *msg.SequenceNumber
```

Comment:
If you're concerned about concurrency, do you think this may be an issue here? Say another goroutine changes `msg.SequenceNumber` to `nil` while this goroutine is going from line 209 to 213, causing a panic. Or is that not really a concern?

Reply:
This is a good point, this nil check is pretty useless without any other concurrency controls. I think in this case we are safe to not use mutexes and drop this check as long as `handleAsync` remains the only goroutine working with the message. Wdyt?

Reply:
I think you're right. I don't think there's a risk of another goroutine changing msg, and I'd rather avoid adding another lock

Reply:
Thanks for the feedback. Will update later tonight 👍

Reply:
Updated